### PR TITLE
feat: home page done

### PR DESCRIPTION
**Pull request by himadri2110**
Features added: 
- Authentication 
- Private & public routes
- Take note input
- Add note via API call

Components:
- NoteInput
- NoteList
- NoteCard

Yet to work on the styling of NoteList and NoteCard
Will add Logout feature later

### CODE REVIEW

**Review by srishti-maurya — APPROVED** [review]
Everything looks good and clean! Great work

File: src/pages/Authentication/Login/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    const { name, value } = e.target;
+    setLogin({ ...login, input: { ...login.input, [name]: value } });
+  };
+
+  const loginHandler = async (e) => {
+    e.preventDefault();
+
+    try {
+      const { data } = await loginService(login.input);
+      localStorage.setItem("isAuth", true);
```
Comment (poornima-alugubelly) [inline src/pages/Authentication/Login/Login.jsx:26]: I didn't know that we could destructure here itself nice 👍

**Comment by poornima-alugubelly** [comment]
Everything is working as expected good to merge

**Review by poornima-alugubelly — APPROVED** [review]

File: src/components/NoteCard/NoteCard.css [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+.note-card .card-action {
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+}
+
+.note-card .card-action button {
+  width: max-content;
+  margin: 0 0.5rem;
```
Comment (msdeshmukh009) [inline src/components/NoteCard/NoteCard.css]: You could've also done this by adding a gap.
```css
.note-card .card-action {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  gap:0.5rem
}
````

**Review by msdeshmukh009 — APPROVED** [review]
This is looking really nice. Maybe you can add a check so that the user doesn't submit empty note.

File: src/components/NoteCard/NoteCard.css (hunk 2 above)
Comment (himadri2110) [reply src/components/NoteCard/NoteCard.css]: Yes, done. Thank you.

**Comment by himadri2110** [comment]
> This is looking really nice. Maybe you can add a check so that the user doesn't submit empty note.

Thank you. Yes, I forgot to put an empty check. 
Thanks for pointing it out!

**Pull request merged**